plugins/filter_kubernetes/kube_meta.c

```diff
@@ -87,14 +89,152 @@ static int file_to_buffer(const char *path,
     return 0;
 }
 
+#ifdef FLB_HAVE_KUBE_TOKEN_COMMAND
+/* Run command to get Kubernetes authorization token */
+static int get_token_with_command(const char *command, 
+                                  char **out_buf, size_t *out_size)
+{
+    FILE *fp;
+    char buf[FLB_KUBE_TOKEN_BUF_SIZE];
+    char *temp;
+    char *res;
+    size_t size = 0;
+    size_t len = 0;
+
+    fp = popen(command, "r");
+    if (fp == NULL) {
+        return -1;
+    }
+
+    res = flb_calloc(1, FLB_KUBE_TOKEN_BUF_SIZE);
+    if (!res) {
+        flb_errno();
+        fclose(fp);
+        return -1;
+    }
+    
+    while (fgets(buf, sizeof(buf), fp) != NULL) {
+        len = strlen(buf);
+        if (len >= FLB_KUBE_TOKEN_BUF_SIZE - 1) {
+            temp = flb_realloc(res, (FLB_KUBE_TOKEN_BUF_SIZE + size) * 2);
+            if (temp == NULL) {
+                flb_errno();
+                flb_free(res);
+                fclose(fp);
+                return -1;
+            }
+            res = temp;
+        }
+        strcpy(res + size, buf);
+        size += len;
+    }
+
+    if (strlen(res) < 1) {
+        flb_free(res);
+        fclose(fp);
+        return -1;
+    }
+    
+    fclose(fp);
+
+    *out_buf = res;
+    *out_size = strlen(res);
+
+    return 0;
+}
+#endif 
+
+/* Set K8s Authorization Token and get HTTP Auth Header */
+static int get_http_auth_header(struct flb_kube *ctx) 
+{
+    int ret;
+    char *temp;
+    char *tk = NULL;
+    size_t tk_size = 0;
+    
+    if (ctx->kube_token_command != NULL) {
+#ifdef FLB_HAVE_KUBE_TOKEN_COMMAND
+        ret = get_token_with_command(ctx->kube_token_command, &tk, &tk_size);
+#else
```

Comment:
can we get ret init as 0 so we won't need a else to define the ret value.

Reply:
I think we should return -1 here as the running command fails and we cannot get token

Reply:
do you mean init it to -1?